packages/santa_close_frame/src/components/ProtectedRoute.tsx

```diff
@@ -0,0 +1,20 @@
+import {FC, ReactElement} from 'react'
+import {Navigate} from 'react-router-dom'
+
+interface ProtectedRouteProps {
+  isAllowed: boolean
+  redirectTo: string
+  children: ReactElement
+}
+
+export const ProtectedRoute: FC<ProtectedRouteProps> = ({
+  isAllowed,
+  redirectTo,
+  children,
+}) => {
+  if (!isAllowed) {
+    return <Navigate replace to={redirectTo} />
+  }
+
+  return children
+}
```

Comment:
```suggestion
export const ProtectedRoute: FC<ProtectedRouteProps> = ({
  isAllowed,
  redirectTo,
  element,
  path
}) => {
  return (
    <Route
      path={path}
      element={isAllowed ? element : <Navigate replace to={redirectTo} />}
    />
}
```
이런 구현은 어떨까요??
사용하는 곳에서 아래처럼 쓸 수 있을 것 같아요!
```tsx
 <Routes>
  <Route path="/" element={<MapAppContainer />} />
  <Route path="/login" element={<Login />} />
  <ProtectedRoute
    path="/protectePath"
    isAllowed={isAvailable} 
    redirectTo="/login"
    element={
        <div>Protected Page</div>
    }
  />
</Routes>
```